syntax = "proto3";

option java_multiple_files = true;
option java_package = "net.xzos.upgradeall.core.route";
option java_outer_classname = "RouteProto";

package routeguide;

service UpdateServerRoute {
  rpc GetReleaseInfo(AppInfo) returns (ReturnValue) {}
}

message AppInfo {
  string hub_uuid = 1;
  repeated AppInfoItem app_info = 2;
}

message AppInfoItem {
  string key = 1;
  string value = 2;
}

message ReturnValue {
  bool valid_hub_uuid = 1;
  bool valid_app = 2;
  repeated ReleaseInfoItem release_info = 3;
}

message ReleaseInfoItem {
  string version_number = 1;
  string change_log = 2;
  repeated AssetItem assets = 3;
}

message AssetItem {
  string file_name = 1;
  string download_url = 2;
  string file_type= 3;
}
